### ISSUE TITLE
Reposition elements screen does not match pdf output

### ISSUE THREAD

**Issue by mattbouchard** [markn86/moodle-mod_customcert#222]
Just as the title suggests, I have laid out my certificate according to what is shown in the "Reposition elements" page, and when I preview it as a pdf, it looks completely different. I have reloaded the page and cleared the cache but there is no change.
![tdg-101-certificateimage](https://user-images.githubusercontent.com/2067365/43661055-0d3ee61a-971e-11e8-8db7-dc44234924b7.PNG)

[Certificate_for_TDG_101.pdf](https://github.com/markn86/moodle-mod_customcert/files/2258516/Certificate_for_TDG_101.pdf)

**Comment by markn86** [comment]
Thanks for the report. Are you able to tell me the dimensions of the page you are using?

**Comment by mattbouchard** [comment]
Thanks so much for responding! Love your plugin.

Width: 270
Height 210

If you have any interest, I can try removing elements to see if the alignment corrects. I also noticed something funny. When I added the signature image (near the time the error was noticed), I placed the top "line image" and made note of the X position (to properly align the bottom "line image"). When I entered it and saved it (on the reposition screen), the line didn't move and the value remained the same. Not sure if it's related. Thanks again for your help.

**Comment by markn86** [comment]
Yes, as much information as possible to reproduce the issue would be very beneficial!

**Comment by mattbouchard** [comment]
Ok. I'll work on something more detailed.

**Comment by mattbouchard** [comment]
Sorry for the delay. There are many problems happening at once, so I will attempt to separate them. For the  moment, let's focus on the first page.

1) There was a minor spacing problem with the bullets, so I removed them from the paragraph above and put them in a separate box. I left in the HTML for an unordered list (ul), and it works fine, looking right in Reposition elements and pdf

2) There was an alignment problem with the address in the bottom left corner. There was some html in there (break tags to get the lines right). When I removed the html and set the width, the problem corrected itself. However, I couldn't get the lines right again, so I just did each line as a separate element. Problem mostly solved. Old spacing was x: 60, Reference point Center. If I add the break tag back in and reset to those coordinates the problem (that it looks right in the Reposition but wrong in the pdf) returns.

3) The white lines on the right side are intended to be signature lines. In the Reposition elements view, the lines are properly sized (at least according to the eye of my graphic designer who made the images for me) and aligned (w: 0, h: 0, X: 160, y: 144) The real size of the image is 292px by 4 px. In the pdf I posted, you can see that the version of the image there is much wider than 4px. At first, when I entered a width of 50, there was no change and the value wasn't saved (despite hitting the save button), and this was on the Reposition elements view. In the Edit Certificate view, if I change the height to 1, the Reposition elements view looks closer to the pdf when generated but is still incorrect.

4) My next work will be on the signature image itself, but I imagine it's the same problem (some kind of resizing happening). I'll post here with more. Let me know if you have questions.

**Comment by adamjacobi** [comment]
Also following, because I'm having the same problem, but I sadly don't have the personal bandwidth to chase after a solution. I'm hoping this is fixed in the next update.  I can PM someone from this project access to my site if you need it to provide more data, though.

**Comment by mattbouchard** [comment]
Welcome, Adam! Did you try setting the width of any images?

**Comment by mattbouchard** [comment]
Interesting! I too encountered some spacing problems when I put in HTML (which is how you did the link, I assume). My workaround was to just break them into separate entities and space them by hand. Here's hoping the plugin maker has a better solution! Good luck!

**Comment by adamjacobi** [comment]
Hi @mattbouchard and @markn86 (tagging, so you know I found the culprit)

I figured it out! Your observation about the link got me thinking this morning. I actually did not put the link there. However, I realized [auto-linking](https://docs.moodle.org/35/en/Auto-linking) was turned on, and it was auto-linking to an activity -- itself! The good news is investigating the issue got me to turn off that and other filters I don't need, which will make server load more efficient.

However, I bet the developer would appreciate hearing that was the culprit (at least in my case).

**Comment by mattbouchard** [comment]
Great news, @adamjacobi ! I was wondering how it all turned out. @markn86 has been busy recently (I assume), so it's nice that we can solve problems together! Great notes about those filters. I didn't even know we had those.

**Comment by timbahula-horizon** [comment]
@mattbouchard , I'm wondering if this is related to Issue #93 . Mark couldn't reproduce my problem and I gave up using the graphical positioning system. It seemed to be a problem with landscape layouts.

**Comment by markn86** [comment]
The auto-linking issue should be fixed now as the format_text/string calls now use the module context, so if it isn't enabled in the module context then the filter should not be applied. See #200.
